benchmark/src/run.ts

```diff
@@ -32,13 +41,27 @@ function benchmark(): void {
 }
 benchmark();
 
+function sortByName({ benchmarkName: a }: BenchmarkResult, { benchmarkName: b }: BenchmarkResult): number {
+    if (a < b) return -1;
+    if (a > b) return 1;
+    return 0;
+}
+
 function compareBenchmarks(oldResults: BenchmarkResult[], newResults: BenchmarkResult[]): ComparisonInfo {
-    const oldResultsMemory = oldResults.filter(isMemoryBenchmarkResult);
-    const newResultsMemory = newResults.filter(isMemoryBenchmarkResult);
+    const oldResultsMemory = oldResults.filter(isMemoryBenchmarkResult).sort(sortByName);
+    const newResultsMemory = newResults.filter(isMemoryBenchmarkResult).sort(sortByName);
 
     const memoryComparisonInfo = compareMemoryBenchmarks(oldResultsMemory, newResultsMemory);
 
-    return { summary: memoryComparisonInfo.summary, text: memoryComparisonInfo.text };
+    const oldResultsRuntime = oldResults.filter(isRuntimeBenchmarkResult).sort(sortByName);
+    const newResultsRuntime = newResults.filter(isRuntimeBenchmarkResult).sort(sortByName);
+
+    const runtimeComparisonInfo = compareRuntimeBenchmarks(oldResultsRuntime, newResultsRuntime);
+
+    return {
```

Comment:
I believe this object is what is base64 encoded and sent to benchviz, it makes sense that just adding a newline breaks this. I'm okay with breaking benchviz for now, but we should make this a nice return type, maybe returning `ComparisonInfo[]` instead? (and add a benchmark name member), or maybe comparisonInfo should have some array internally? I want an array somewhere, no \n concatenation.

Reply:
That depends on how benchviz is meant to work. Could we just return the JSON result or something similar, and have benchviz format it? Although, the summary markdown text is useful for local viewing

Reply:
This object as JSON is what is visualized by benchviz. Really this should just be the data, any visualization like benchviz or even printing it out as markdown should probably take this thing as input and display it however it sees fit.

Reply:
I'm thinking of creating an interface for "Benchmark result summary" as an intermediary, that gets both sent to benchviz and also is able to be formatted to markdown. Would that work?

Reply:
Also, I think the online benchviz is breaking for a different reason: I'm getting `ReferenceError: Buffer is not defined` in the console

Reply:
Yeah I think so too, it seems broken for another reason, probably to do with our latest docusaurus update. I'm fine with keeping it broken, but like I said I don't want it to have to figure out what to show based on a \n somewhere in there. I propose something like this:
```ts
return {
    [BenchmarkKind.Memory]: {
        summary: memoryComparisonInfo.summary,
        text: memoryComparisonInfo.text,
    },
    [BenchmarkKind.Runtime]: {
        summary: runtimeComparisonInfo.summary,
        text: runtimeComparisonInfo.text,
    },
}
```
And adjusting return type to `Record<BenchmarkKind, ComparisonInfo>`, we can fix benchviz later.

Reply:
I made a separate PR on typescripttolua.github.io that fixes benchviz and adds support for runtime.

I'll make a separate commit with this proposal for the markdown output.